src/axolotl/kernels/memft_xentropy.py

```
@@ -0,0 +1,214 @@
+
+class MemFTLinearCrossEntropy(torch.autograd.Function):
+    """Fused linear + MemFT-OT cross-entropy.
+
+    forward(hidden [N, H], weight [V, H], labels [N]) -> scalar loss.
+    """
+
+    @staticmethod
+    def forward(
+        ctx, hidden, weight, labels, critical_loss, epsilon, ignore_index, chunk_tokens
+    ):
+        n_tokens, hidden_size = hidden.shape
```

Ruff (0.15.15)

[warning] Unpacked variable `hidden_size` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)